can/interfaces/multicast_ip/bus.py

```diff
@@ -0,0 +1,338 @@
+                         guaranteed to be closed/cleaned up
+        """
+        # create the UDP socket
+        # this might already fail but then there is nothing to clean up
+        sock = socket.socket(address_family, socket.SOCK_DGRAM)
+
+        # configure the socket
+        try:
+
+            # set hop limit / TTL
+            ttl_as_binary = struct.pack("@I", self.hop_limit)
+            if self.ip_version == 4:
+                sock.setsockopt(
+                    socket.IPPROTO_IP, socket.IP_MULTICAST_TTL, ttl_as_binary
+                )
+            else:
+                sock.setsockopt(
+                    socket.IPPROTO_IPV6, socket.IPV6_MULTICAST_HOPS, ttl_as_binary
+                )
+
+            # Allow multiple programs to access that address + port
+            sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+
+            # set how to receive timestamps
+            sock.setsockopt(socket.SOL_SOCKET, SO_TIMESTAMPNS, 1)
+
+            # Bind it to the port (ony any interface)
+            sock.bind(("", self.port))
+
+            # Join the multicast group
+            group_as_binary = socket.inet_pton(address_family, self.group)
+            if self.ip_version == 4:
+                request = group_as_binary + struct.pack("@I", socket.INADDR_ANY)
+                sock.setsockopt(socket.IPPROTO_IP, socket.IP_ADD_MEMBERSHIP, request)
+            else:
+                request = group_as_binary + struct.pack("@I", 0)
+                sock.setsockopt(socket.IPPROTO_IPV6, socket.IPV6_JOIN_GROUP, request)
+
+            return sock
+
+        except OSError as error:
+            # clean up the incompletely configured but opened socket
+            try:
+                sock.close()
+            except OSError as close_error:
+                # ignore but log any failures in here
+                log.warning("Could not close partly configured socket: %s", close_error)
+
+            # still raise the error
+            raise error
+
+    def send(
+        self,
+        data: ReadableBytesLike,
+        timeout: Optional[float] = None,
+        start: int = 0,
+        count: Optional[int] = None,
+    ) -> None:
+        """Send this data to all participants. This call blocks.
+
+        If `count` is set to an integer, only so many bytes beginning with `start` (inclusive) are sent.
+
+        :param timeout: the timeout in seconds after which an Exception is raised is sending has failed
+        :param data: the data to be sent
+        :param start: where to start the slice to be sent
+        :param count: how many bytes to send or `None` to send everything
+        :raises OSError: if an error occurred while writing to the underlying socket
+        :raises socket.timeout: if the timeout ran out before sending was completed (this is a subclass of
+                                *OSError*)
+        """
+        if not count:
```

Comment:
A strategy I've been applying a lot lately is "inverse the if not". In this case, I would change `if not count` into `if count` and swap the arms of the if. Non-negative logic is often easier to read.

Reply:
Yep. Actually, the `start` and `count` parameters were never really used, so I just deleted that option. Now its simpler too. :P